jdaviz/core/helpers.py

```diff
@@ -402,6 +407,88 @@ def show_in_new_tab(self, title=None):  # pragma: no cover
 
+    def get_data(self, data_label=None, cls=None, subset_to_apply=None, statistic=None):
+        """
+        Returns data with name equal to data_label of type cls with subsets applied from
+        subset_to_apply.
+
+        Parameters
+        ----------
+        data_label : str, optional
+            Provide a label to retrieve a specific data set from data_collection.
+        cls : `~specutils.Spectrum1D`, `~astropy.nddata.CCDData`, optional
+            The type that data will be returned as.
+        subset_to_apply : str, optional
+            Subset that is to be applied to data before it is returned.
+        statistic : {'minimum', 'maximum', 'mean', 'median', 'sum'}, optional
+            The statistic to use to collapse the dataset.
+
+        Returns
+        -------
+        data : cls
+            Data is returned as type cls with subsets applied.
+
+        """
+        list_of_valid_statistic_values = ['minimum', 'maximum', 'mean',
+                                          'median', 'sum']
+        if statistic and statistic not in list_of_valid_statistic_values:
+            raise ValueError(f"statistic {statistic} not in list of valid"
+                             f" statistic values {list_of_valid_statistic_values}")
+
+        if data_label and data_label not in self.app.data_collection.labels:
+            raise ValueError(f'{data_label} not in {self.app.data_collection.labels}.')
+        elif not data_label and len(self.app.data_collection) > 1:
+            raise ValueError('data_label must be set if more than'
+                             ' one data exists in data_collection.')
+        elif not data_label and len(self.app.data_collection) == 1:
+            data_label = self.app.data_collection[0].label
+
+        if cls is not None and not isclass(cls):
+            raise TypeError(
+                "cls in get_data must be a class or None.")
+        data = self.app.data_collection[data_label]
+
+        if not cls:
+            if len(data.shape) == 2:
+                cls = CCDData
+            elif len(data.shape) in [1, 3]:
+                cls = Spectrum1D
+        if not subset_to_apply:
+            if 'Trace' in data.meta:
+                data = data.get_object()
+            elif cls == Spectrum1D:
+                data = data.get_object(cls=cls, statistic=statistic)
+            else:
+                data = data.get_object(cls=cls)
+
+            return data
+
+        if not cls and subset_to_apply:
+            raise AttributeError(f"A valid cls must be provided to"
+                                 f" apply subset {subset_to_apply} to data. "
+                                 f"Instead, {cls} was given.")
+
+        # Loop through each subset
+        for subsets in self.app.data_collection.subset_groups:
+            # If name matches the name in subsets_to_apply, continue
+            if subsets.label.lower() == subset_to_apply.lower():
```

Comment:
if `subset_to_apply` is not valid, an error should be raised earlier (now it still returns the glue object in specviz when passing an invalid string to `subset_to_apply`)